apollo-router/src/error.rs

```diff
@@ -160,7 +160,7 @@ impl FetchError {
 impl ErrorExtension for FetchError {
     fn extension_code(&self) -> String {
         match self {
-            FetchError::ValidationInvalidTypeVariable { .. } => "VALIDATION_INVALID_TYPE_VARIABLE",
+            FetchError::ValidationInvalidTypeVariable { .. } => "GRAPHQL_VALIDATION_FAILED",
```

Comment:
These should be separate error codes. `GRAPHQL_VALIDATION_FAILED` means that the operation was invalid, which doesn't account for the actual values provided to variables (spec-wise that is part of execution and not validation). In apollo-server, `BAD_USER_INPUT` is used when variable input coercion fails, including wrong types or non-existent enum values, but maybe we should continue to use our value for back compat?

Reply:
apparently the gateway returns `GRAPHQL_VALIDATION_FAILED` here https://github.com/apollographql/router/issues/4633

Reply:
The gateway raises an entirely different error there (an input type doesn't exist in the schema), it never gets to input validation

Reply:
Oh, it might be that #4633 should indeed raise a validation error, but because with the JS query planner the router ends up doing input coercion *before* graphql validation, this error is the one that gets returned. So #4551 might solve the difference.

Reply:
so which error should the router return exactly?

Reply:
I was wrong before about #4663, as you said the error on router comes from the subgraph, so there is also something else going on. I still suspect the reproduction may be wrong, or there is something very weird as both operations are validated with JS validation and should thus have the same result.

For this PR, I think we should continue to use the `VALIDATION_INVALID_TYPE_VARIABLE` error code. We could also rename to `BAD_USER_INPUT` to match gateway but people could be relying on our error code already.

Reply:
agreed, I changed it back